pkg/codegen/utilities.go

```diff
@@ -79,3 +82,31 @@ func SortedKeys(m interface{}) []string {
 
 	return keys
 }
+
+// CleanDir removes all existing files from a directory except those in the exclusions list.
+func CleanDir(dirPath string, exclusions *[]string) error {
```

Comment:
Let's just pass in the set itself here. It's safe to pass `nil` to indicate an empty set.

```suggestion
func CleanDir(dirPath string, exclusions codegen.StringSet) error {
```